internal/provider/resource_engine_configuration.go

```diff
@@ -149,6 +196,171 @@ func resourceEngineConfiguration() *schema.Resource {
 				Type:     schema.TypeString,
 				Computed: true,
 			},
+			"device_type": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice([]string{BLOCK, OBJECT}, false),
+			},
+			"object_storage_params": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"region": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"bucket": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"endpoint": {
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"size": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validateStorageSize,
+						},
+						"auth_type": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      ROLE,
+							ValidateFunc: validation.StringInSlice([]string{ROLE, ACCESS_KEY}, false),
+						},
+						"s3_instance_profile": {
```

Comment:
I am ok with this but can you think of making 
```
Default:  "S3ObjectStoreAccessInstanceProfile",
```
and later in the handling part make it as 
```
} else {
    params.S3_INSTANCE_PROFILE = object_storage_params[0].(map[string]interface{})["s3_instance_profile"].(string)
}
```